File: typescript-client-sdk/src/sdk/models/operations/deletepet.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import * as utils from "../utils";
import { Type } from "class-transformer";


export class DeletePetPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: number;
}

export class DeletePetRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DeletePetPathParams;

  @SpeakeasyMetadata()
  retries?: utils.RetryConfig;
}

export class DeletePetResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/deletepet.go
package operations

import (
	"github.com/speakeasy-sdks/nolan-test/pkg/models/shared"
	"github.com/speakeasy-sdks/nolan-test/pkg/utils"
)

type DeletePetPathParams struct {
	ID int64 `pathParam:"style=simple,explode=false,name=id"`
}

type DeletePetRequest struct {
	PathParams DeletePetPathParams
	Retries    *utils.RetryConfig
}

type DeletePetResponse struct {
	ContentType string
	Error       *shared.Error
	StatusCode  int
}
